File: src/optical_element/MATLAB/Furion_paraboid_collimation_Mirror.m
classdef Furion_paraboid_collimation_Mirror<w_oe
    %波动光学圆柱镜  
    properties
        r; %圆柱镜面曲率半径  
        r1;
  
    end
    
    methods
        function obj=Furion_paraboid_collimation_Mirror( beam_in,ds,di,chi,theta,surface,r1,grating)    
            obj = obj@w_oe( beam_in,ds,di,chi,theta,surface,grating) ;  %初始化父类  optical_element的参数      
           obj.r1=r1; %物距
      
            obj.reflect;%反射光束          
        end
    end   
    
      methods %面型反射,追迹
        function tracing(obj)
           obj.g_mirror = g_Furion_paraboid_collimation_Mirror( obj.gbeam_in,obj.ds,obj.di,obj.chi,obj.theta,obj.surface,obj.r1,obj.grating)   ;
        end
      end
    
end

File: src/optical_element/Furion_paraboid_collimation_Mirror.cpp
#include "Furion_paraboid_collimation_Mirror.h"

using namespace Furion_NS;

Furion_Paraboid_Collimation_Mirror::Furion_Paraboid_Collimation_Mirror(Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r1, Grating* grating)
    : W_Oe(beam_in, ds, di, chi, theta, surface, grating)
{
    cout << "Furion_Paraboid_Collimation_Mirror 初始化" << endl;
    this->r1 = r1;
}

Furion_Paraboid_Collimation_Mirror::~Furion_Paraboid_Collimation_Mirror()
{

}

void Furion_Paraboid_Collimation_Mirror::run(Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, Grating* grating)
{
    cout << "Furion_Paraboid_Collimation_Mirror的run" << endl;
    reflect(beam_in, ds, di, chi, theta);
}

std::string Furion_Paraboid_Collimation_Mirror::tracing()
{
    cout << "Furion_Paraboid_Collimation_Mirror的tracing" << endl;
    this->g_FPC_mirror = new G_Furion_Paraboid_Collimation_Mirror(this->gbeam_in, this->ds, this->di, this->chi, this->theta, this->surface, this->r1, this->grating);
    g_FPC_mirror->run(this->gbeam_in, this->ds, this->di, this->chi, this->theta, this->surface, this->r1, this->grating);

    return ("Furion_Paraboid_Collimation_Mirror");
}